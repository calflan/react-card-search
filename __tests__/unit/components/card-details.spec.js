import React from 'react';
import CardDetails from '../../../src/js/components/card-details';

describe('Contact Component', () => {
  let component;
  let details;
  let mockToggleCardExpandCallBack = jest.fn();

  beforeEach(() => {
    details = {
      Title: "New Job Card",
      ImageUrls: [
        {
          ImageUrl: "http://someimage.com"
        },
        {
          ImageUrl: "http://anotherimage.com"
        }
      ],
      AvailableSizes: [
        {
          DisplayName: "Large"
        },
        {
          DisplayName: "Medium"
        }
      ],
    };

    component = shallow(<CardDetails details={details} status="complete" options={{isOpen: true, detailsExpanded: mockToggleCardExpandCallBack }}/>);
  });

  test('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should close card details and call detailsExpanded callback and _handleClose method when close is clicked', () => {
    const handleClose = jest.spyOn(component.instance(), '_handleClose');
    const closeIcon = component.find('.card-listings__card-details-close');

    closeIcon.simulate('click');
    component.instance()._handleClose();
    
    expect(handleClose).toHaveBeenCalled();
    expect(mockToggleCardExpandCallBack).toHaveBeenCalled();
  });
});