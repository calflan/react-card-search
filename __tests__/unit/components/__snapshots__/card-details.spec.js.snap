// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component should match snapshot 1`] = `
<article
  className="card-listings__card-details "
  role="dialog"
>
  <a
    className="card-listings__card-details-overlay"
    onClick={[Function]}
    role="overlay"
  />
  <div
    className="card-listings__card-details-inner"
  >
    <a
      className="card-listings__card-details-close"
      href="#"
      onClick={[Function]}
      role="closelink"
    >
      X
    </a>
    <img
      alt="New Job Card"
      role="productimage"
      src="http://someimage.com"
    />
    <section
      className="card-listings__card-details-inner-information"
    >
      <h3
        role="heading"
      >
        New Job Card
      </h3>
      <h4
        className="card-listings__card-details-stock-status--outOfStock"
        role="productstockstatus"
      >
        Sorry, this product is out of stock!
      </h4>
      <ul>
        <h4
          role="productsizesheading"
        >
          Sizes available
        </h4>
        <li
          key="0"
          role="productsizelistitem"
        >
          Large
           - £
        </li>
        <li
          key="1"
          role="productsizelistitem"
        >
          Medium
           - £
        </li>
      </ul>
      <h4
        role="descriptionheading"
      >
        Description
      </h4>
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
        role="description"
      />
    </section>
  </div>
</article>
`;
